addons/delivery/models/delivery_carrier.py

```diff
@@ -231,7 +231,7 @@ def rate_shipment(self, order):
             res['carrier_price'] = res['price']
             # free when order is large enough
             amount_without_delivery = order._compute_amount_total_without_delivery()
-            if res['success'] and self.free_over and self._compute_currency(order, amount_without_delivery, 'pricelist_to_company') >= self.amount:
+            if res['success'] and (self.free_over and self.delivery_type == 'fixed') and self._compute_currency(order, amount_without_delivery, 'pricelist_to_company') >= self.amount:
```

Comment:
The view is hiding this field only in the case of `base_on_rule`. The condition is still valid for external providers.
```suggestion
            if res['success'] and (self.free_over and self.delivery_type != 'base_on_rule') and self._compute_currency(order, amount_without_delivery, 'pricelist_to_company') >= self.amount:
```
(+ this line is getting very long, you may want to split it)

Reply:
Thanks for the review :)
Targeting 16.3 instead of 16.2 because it is no longer supported and trying to target it causes problems with upgrade scripts.

Reply:
Yes, 16.2 is not targeted anymore, I should have told you in the first message, sorry. Thanks for the changes. @Feyensv will do the last review. Cheers